CI: tests/arch/x86/info/arch.x86.info.userspace fails to build for acrn_ehl_crb due to pcie.c compilation error

**Describe the bug**
`tests/arch/x86/info/arch.x86.info.userspace` fails to build for `acrn_ehl_crb` as seen in https://github.com/zephyrproject-rtos/zephyr/actions/runs/7859262283/job/21474029951

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -p -b acrn_ehl_crb tests/arch/x86/info -T arch.x86.info.userspace`
2. See error

**Expected behavior**
CI build succeeds.

**Impact**
CI build failure.

**Logs and console output**
```
$ west build -p -b acrn_ehl_crb tests/arch/x86/info -T arch.x86.info.userspace
-- west build: making build dir /home/brix/Projects/zephyrproject/zephyr/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/brix/Projects/zephyrproject/zephyr/tests/arch/x86/info
-- CMake version: 3.28.1
-- Found Python3: /home/brix/Projects/zephyrproject/venv/bin/python3 (found suitable version "3.10.12", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /home/brix/.cache/zephyr
-- Zephyr version: 3.6.0-rc2 (/home/brix/Projects/zephyrproject/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: acrn_ehl_crb
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.5 (/opt/zephyr-sdk/0.16.5)
-- Found toolchain: zephyr 0.16.5 (/opt/zephyr-sdk/0.16.5)
-- Found Dtc: /opt/zephyr-sdk/0.16.5/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/brix/Projects/zephyrproject/zephyr/boards/x86/acrn/acrn_ehl_crb.dts
-- Found devicetree overlay: /home/brix/Projects/zephyrproject/zephyr/tests/arch/x86/info/app.overlay
-- Generated zephyr.dts: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /home/brix/Projects/zephyrproject/zephyr/build/zephyr/dts.cmake
Parsing /home/brix/Projects/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/brix/Projects/zephyrproject/zephyr/boards/x86/acrn/acrn_ehl_crb_defconfig'
Merged configuration '/home/brix/Projects/zephyrproject/zephyr/tests/arch/x86/info/prj.conf'
Merged configuration '/home/brix/Projects/zephyrproject/zephyr/build/zephyr/misc/generated/extra_kconfig_options.conf'
Configuration saved to '/home/brix/Projects/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- Found GnuLd: /opt/zephyr-sdk/0.16.5/x86_64-zephyr-elf/bin/../lib/gcc/x86_64-zephyr-elf/12.2.0/../../../../x86_64-zephyr-elf/bin/ld.bfd (found version "2.38") 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/zephyr-sdk/0.16.5/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc
-- Using ccache: /usr/bin/ccache
-- Configuring done (5.3s)
-- Generating done (0.2s)
-- Build files have been written to: /home/brix/Projects/zephyrproject/zephyr/build
-- west build: building application
[  0%] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[  1%] Built target parse_syscalls_target
[  2%] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[  2%] Built target kobj_types_h_target
[  2%] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[  2%] Built target syscall_list_h_target
[  2%] Generating include/generated/driver-validation.h
[  2%] Built target driver_validation_h_target
[  2%] Building C object zephyr/CMakeFiles/offsets.dir/arch/x86/core/offsets/offsets.c.obj
[  2%] Built target offsets
[  3%] Generating include/generated/offsets.h
[  3%] Built target offsets_h
[  3%] Generating include/generated/version.h
-- Zephyr version: 3.6.0-rc2 (/home/brix/Projects/zephyrproject/zephyr), build: v3.6.0-rc2-50-g61f065230ad9
[  3%] Built target version_h
[  3%] Built target zephyr_generated_headers
[  3%] Building C object CMakeFiles/app.dir/src/main.c.obj
[  3%] Building C object CMakeFiles/app.dir/src/timer.c.obj
[  4%] Building C object CMakeFiles/app.dir/src/acpi.c.obj
[  4%] Building C object CMakeFiles/app.dir/src/multiboot.c.obj
[  4%] Building C object CMakeFiles/app.dir/src/memmap.c.obj
[  5%] Linking C static library app/libapp.a
[  5%] Built target app
[  6%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/heap/heap.c.obj
[  6%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
[  6%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
[  7%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[  7%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[  7%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[  8%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[  8%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/mutex.c.obj
[  8%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/user_work.c.obj
[  9%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/dec.c.obj
[  9%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/hex.c.obj
[  9%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/rb.c.obj
[ 10%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/timeutil.c.obj
[ 10%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/bitarray.c.obj
[ 10%] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[ 11%] Building C object zephyr/CMakeFiles/zephyr.dir/boards/x86/acrn/board_acrn.c.obj
[ 11%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj
[ 11%] Linking C static library libzephyr.a
[ 11%] Built target zephyr
[ 12%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/main_weak.c.obj
[ 12%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/banner.c.obj
[ 12%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/busy_wait.c.obj
[ 13%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/device.c.obj
[ 13%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/errno.c.obj
[ 13%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj
[ 14%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj
[ 14%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/kheap.c.obj
[ 14%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mem_slab.c.obj
[ 15%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj
[ 15%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/version.c.obj
[ 15%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sched.c.obj
[ 16%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj
[ 16%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mailbox.c.obj
[ 16%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/msg_q.c.obj
[ 17%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj
[ 17%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/queue.c.obj
[ 17%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sem.c.obj
[ 18%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/stack.c.obj
[ 18%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/system_work_q.c.obj
[ 18%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/work.c.obj
[ 19%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/condvar.c.obj
[ 19%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/smp.c.obj
[ 19%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timeout.c.obj
[ 20%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timer.c.obj
[ 20%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mmu.c.obj
[ 20%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mempool.c.obj
[ 21%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/futex.c.obj
[ 21%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mem_domain.c.obj
[ 22%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/userspace_handler.c.obj
[ 22%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/userspace.c.obj
[ 22%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/dynamic_disabled.c.obj
[ 23%] Linking C static library libkernel.a
[ 23%] Built target kernel
[ 23%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/cpuhalt.c.obj
[ 24%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/prep_c.c.obj
[ 24%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/fatal.c.obj
[ 24%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/cpuid.c.obj
[ 25%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/spec_ctrl.c.obj
[ 25%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/memmap.c.obj
[ 25%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/pcie.c.obj
[ 26%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/multiboot.c.obj
[ 26%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/legacy_bios.c.obj
[ 26%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/x86_acpi.c.obj
[ 27%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/x86_mmu.c.obj
[ 27%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/userspace.c.obj
[ 27%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/tls.c.obj
[ 28%] Building ASM object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/intel64/locore.S.obj
[ 28%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/intel64/cpu.c.obj
[ 28%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/intel64/irq.c.obj
[ 29%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/intel64/thread.c.obj
[ 29%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/intel64/fatal.c.obj
[ 30%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/intel64/smp.c.obj
[ 30%] Building ASM object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/intel64/userspace.S.obj
[ 30%] Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/intel64/tls.c.obj
[ 31%] Linking C static library libarch__x86__core.a
[ 31%] Built target arch__x86__core
[ 31%] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/libc-hooks.c.obj
[ 31%] Linking C static library liblib__libc__picolibc.a
[ 31%] Built target lib__libc__picolibc
[ 31%] Building C object zephyr/lib/libc/common/CMakeFiles/lib__libc__common.dir/source/stdlib/abort.c.obj
[ 31%] Building C object zephyr/lib/libc/common/CMakeFiles/lib__libc__common.dir/source/stdlib/malloc.c.obj
[ 32%] Linking C static library liblib__libc__common.a
[ 32%] Built target lib__libc__common
[ 32%] Building C object zephyr/lib/acpi/CMakeFiles/lib__acpi.dir/acpi.c.obj
[ 34%] Linking C static library liblib__acpi.a
[ 34%] Built target lib__acpi
[ 35%] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_ioapic.c.obj
[ 35%] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_loapic.c.obj
[ 35%] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_system_apic.c.obj
[ 36%] Linking C static library libdrivers__interrupt_controller.a
[ 36%] Built target drivers__interrupt_controller
[ 36%] Building C object zephyr/drivers/pcie/host/CMakeFiles/drivers__pcie__host.dir/pcie.c.obj
In file included from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/toolchain.h:50,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/__assert.h:11,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/iterable_sections.h:10,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/logging/log_instance.h:10,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/logging/log.h:10,
                 from /home/brix/Projects/zephyrproject/zephyr/drivers/pcie/host/pcie.c:10:
/home/brix/Projects/zephyrproject/zephyr/drivers/pcie/host/pcie.c: In function 'pcie_init':
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/toolchain/gcc.h:87:36: error: static assertion failed: "No HID property for PCIe devicetree node"
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                    ^~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/pcie/host/pcie.c:567:9: note: in expansion of macro 'BUILD_ASSERT'
  567 |         BUILD_ASSERT(ACPI_DT_HAS_HID(DT_DRV_INST(0)),
      |         ^~~~~~~~~~~~
In file included from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/arch/x86/arch.h:9,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/arch/cpu.h:15,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/cbprintf_internal.h:17,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/cbprintf.h:124,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/logging/log_msg.h:11,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/logging/log_core.h:9,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/logging/log.h:11:
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:334:40: error: 'DT_N_INST_0_pcie_controller_P_acpi_hid' undeclared (first use in this function)
  334 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                                        ^~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:4452:29: note: in definition of macro 'DT_CAT3'
 4452 | #define DT_CAT3(a1, a2, a3) a1 ## a2 ## a3
      |                             ^~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/acpi/acpi.h:92:30: note: in expansion of macro 'DT_PROP'
   92 | #define ACPI_DT_HID(node_id) DT_PROP(node_id, acpi_hid)
      |                              ^~~~~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/pcie/host/pcie.c:569:15: note: in expansion of macro 'ACPI_DT_HID'
  569 |         hid = ACPI_DT_HID(DT_DRV_INST(0));
      |               ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:334:31: note: in expansion of macro 'UTIL_CAT'
  334 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:3410:27: note: in expansion of macro 'DT_INST'
 3410 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
      |                           ^~~~~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/pcie/host/pcie.c:569:27: note: in expansion of macro 'DT_DRV_INST'
  569 |         hid = ACPI_DT_HID(DT_DRV_INST(0));
      |                           ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:334:40: note: each undeclared identifier is reported only once for each function it appears in
  334 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                                        ^~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:4452:29: note: in definition of macro 'DT_CAT3'
 4452 | #define DT_CAT3(a1, a2, a3) a1 ## a2 ## a3
      |                             ^~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/acpi/acpi.h:92:30: note: in expansion of macro 'DT_PROP'
   92 | #define ACPI_DT_HID(node_id) DT_PROP(node_id, acpi_hid)
      |                              ^~~~~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/pcie/host/pcie.c:569:15: note: in expansion of macro 'ACPI_DT_HID'
  569 |         hid = ACPI_DT_HID(DT_DRV_INST(0));
      |               ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:334:31: note: in expansion of macro 'UTIL_CAT'
  334 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:3410:27: note: in expansion of macro 'DT_INST'
 3410 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
      |                           ^~~~~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/pcie/host/pcie.c:569:27: note: in expansion of macro 'DT_DRV_INST'
  569 |         hid = ACPI_DT_HID(DT_DRV_INST(0));
      |                           ^~~~~~~~~~~
gmake[2]: *** [zephyr/drivers/pcie/host/CMakeFiles/drivers__pcie__host.dir/build.make:76: zephyr/drivers/pcie/host/CMakeFiles/drivers__pcie__host.dir/pcie.c.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3946: zephyr/drivers/pcie/host/CMakeFiles/drivers__pcie__host.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
FATAL ERROR: command exited with status 2: /home/brix/Projects/zephyrproject/venv/bin/cmake --build /home/brix/Projects/zephyrproject/zephyr/build
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA: 61f065230ad9bca91633cd57b9dcd5e90c5d8bc7

--- a/boards/x86/acrn/acrn.dts
+++ b/boards/x86/acrn/acrn.dts
@@ -27,6 +27,14 @@
 		zephyr,console = &uart0;
 		zephyr,shell-uart = &uart0;
 	};
+
+	pcie0: pcie0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "pcie-controller";
+		acpi-hid = "PNP0A08";
+		ranges;
+	};
 };
 
 &uart0 {

